src/test/regress/expected/foreign_key_to_reference_table.out

```diff
@@ -1357,6 +1355,414 @@ ERROR:  cannot establish a new connection for placement 7000388, since DDL has b
+CREATE TABLE test_table_1(id int PRIMARY KEY, id2 int);
+CREATE TABLE test_table_2(id int PRIMARY KEY, value_1 int, FOREIGN KEY(value_1) REFERENCES test_table_1(id));
+BEGIN;
+  SELECT create_reference_table('test_table_1');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+  SELECT create_distributed_table('test_table_2', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+  ALTER TABLE test_table_2 DROP COLUMN value_1;
+COMMIT;
+SELECT count(*) FROM table_fkeys_in_workers WHERE relid LIKE 'fkey_reference_table.%' AND refd_relid LIKE 'fkey_reference_table.%';
+ count 
+-------
+     0
+(1 row)
+
+DROP TABLE test_table_1, test_table_2;
+-- check if we can drop the column which foreign key is referencing to
+CREATE TABLE test_table_1(id int PRIMARY KEY, id2 int);
+CREATE TABLE test_table_2(id int PRIMARY KEY, value_1 int, FOREIGN KEY(value_1) REFERENCES test_table_1(id));
+SELECT create_reference_table('test_table_1');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table_2', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ALTER TABLE test_table_1 DROP COLUMN id CASCADE;
+NOTICE:  drop cascades to constraint test_table_2_value_1_fkey on table test_table_2
+SELECT count(*) FROM table_fkeys_in_workers WHERE relid LIKE 'fkey_reference_table.%' AND refd_relid LIKE 'fkey_reference_table.%';
+ count 
+-------
+     0
+(1 row)
+
+DROP TABLE test_table_1, test_table_2;
+-- check if we can drop the column which foreign key is referencing from in a transaction block
+CREATE TABLE test_table_1(id int PRIMARY KEY, id2 int);
+CREATE TABLE test_table_2(id int PRIMARY KEY, value_1 int, FOREIGN KEY(value_1) REFERENCES test_table_1(id));
+BEGIN;
+  SELECT create_reference_table('test_table_1');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+  SELECT create_distributed_table('test_table_2', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+  ALTER TABLE test_table_1 DROP COLUMN id CASCADE;
+NOTICE:  drop cascades to constraint test_table_2_value_1_fkey on table test_table_2
+COMMIT;
+SELECT count(*) FROM table_fkeys_in_workers WHERE relid LIKE 'fkey_reference_table.%' AND refd_relid LIKE 'fkey_reference_table.%';
+ count 
+-------
+     0
+(1 row)
+
+DROP TABLE test_table_1, test_table_2;
+-- check if we can alter the column type which foreign key is referencing to
+CREATE TABLE test_table_1(id int PRIMARY KEY, id2 int);
+CREATE TABLE test_table_2(id int PRIMARY KEY, value_1 int, FOREIGN KEY(value_1) REFERENCES test_table_1(id));
+SELECT create_reference_table('test_table_1');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table_2', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO test_table_1 VALUES (1,1), (2,2), (3,3);
```

Comment:
Can you try a case there `SET DATA TYPE` fails since say there is already an int which is `MAX_INT_64 - 1`